Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -323,14 +320,184 @@ def tricontourf(self,*args,**kwargs):
         
         For keywords related to workspaces, see :func:`mantid.plots.functions.tricontourf`             
         '''
-        if self.validate_args(*args):
-            mantid.kernel.logger.debug('using mantid.plots.functions')     
-            return mantid.plots.functions.tricontourf(self,*args,**kwargs)
+        if mantid.plots.helperfunctions.validate_args(*args):
+            mantid.kernel.logger.debug('using mantid.plots.plotfunctions')
+            return mantid.plots.plotfunctions.tricontourf(self, *args, **kwargs)
         else:
-            return Axes.tricontourf(self,*args,**kwargs)
+            return Axes.tricontourf(self, *args, **kwargs)
 
 
-register_projection(MantidAxes)
+class MantidAxes3D(Axes3D):
+    '''
+    This class defines the **mantid3d** projection for 3d plotting. One chooses
+    this projection using::
+
+        import matplotlib.pyplot as plt
+        from mantid import plots
+        fig, ax = plt.subplots(subplot_kw={'projection':'mantid3d'})
+
+    or::
+
+        import matplotlib.pyplot as plt
+        from mantid import plots
+        fig = plt.figure()
+        ax = fig.add_subplot(111,projection='mantid3d')
+
+    The mantid3d projection allows replacing the array objects with mantid workspaces.
+    '''
+
+    name = 'mantid3d'
+
+    def plot(self, *args, **kwargs):
+        '''
+        If the **mantid3d** projection is chosen, it can be
+        used the same as :py:meth:`matplotlib.axes.Axes3D.plot` for arrays,
+        or it can be used to plot :class:`mantid.api.MatrixWorkspace`
+        or :class:`mantid.api.IMDHistoWorkspace`. You can have something like::
+
+            import matplotlib.pyplot as plt
+            from mantid import plots
+
+            ...
+
+            fig, ax = plt.subplots(subplot_kw={'projection':'mantid3d'})
+            ax.plot(workspace) #for workspaces
+            ax.plot(x,y,z)     #for arrays
+            fig.show()
+
+        For keywords related to workspaces, see :func:`mantid.plots.functions.plot3D`
+        '''
+        if mantid.plots.helperfunctions.validate_args(*args):
+            mantid.kernel.logger.debug('using mantid.plots.plotfunctions3D')
+            return mantid.plots.plotfunctions3D.plot3d(self, *args, **kwargs)
```

Comment:
For consistency with matplotlib, this function should call `mantid.plots.plotfunctions3D.plot`, so you should rename your `plot3d`